### PR TITLE
Add IsSupported() API which also parses /proc/self/status

### PR DESCRIPTION
**Pull request by saschagrunert**
We can use `/proc/self/status` to pre-check if seccomp is available on the
local system. This also means that we can call this function without
having a need for the `seccomp` build tag. If the build tag is available,
then the `IsEnabled()` function will now re-use the new `IsSupported()`
API.

### CODE REVIEW

**Comment by pcmoore** [comment]
Related, especially the most recent comments: https://github.com/seccomp/libseccomp/issues/45

**Comment by rhatdan** [comment]
@vrothberg @giuseppe PTAL

File: seccomp_linux.go [hunk 1]
```diff
@@ -180,12 +179,5 @@ func createSpecsSyscall(names []string, action Action, args []*Arg, errnoRet *ui
...
 func IsEnabled() bool {
-	// Check if Seccomp is supported, via CONFIG_SECCOMP.
-	if err := unix.Prctl(unix.PR_GET_SECCOMP, 0, 0, 0, 0); err != unix.EINVAL {
-		// Make sure the kernel has CONFIG_SECCOMP_FILTER.
-		if err := unix.Prctl(unix.PR_SET_SECCOMP, unix.SECCOMP_MODE_FILTER, 0, 0, 0); err != unix.EINVAL {
-			return true
-		}
-	}
-	return false
+	return IsSupported()
```
Comment (vrothberg) [inline seccomp_linux.go:182]: I think we could make `isSupported` private instead?

File: supported.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+	// try to read from status for kernels > 3.8
+	status, err := parseStatusFile(statusFilePath)
+	if err == nil {
+		_, ok := status["Seccomp"]
+		return ok
+	}
+
+	// check if Seccomp is supported via CONFIG_SECCOMP
+	if err := unix.Prctl(unix.PR_GET_SECCOMP, 0, 0, 0, 0); !errors.Is(err, unix.EINVAL) {
```
Comment (vrothberg) [inline supported.go:36]: I wonder if this check first is faster?

File: supported_test.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
+package seccomp
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
```
Comment (vrothberg) [inline supported_test.go:10]: Could we add an entire `status` file as we do in https://github.com/containers/psgo/blob/master/internal/proc/status_test.go?

File: supported.go (hunk 2 above)
Comment (saschagrunert) [reply supported.go:36]: Probably, let's turn them around.

File: supported_test.go (hunk 3 above)
Comment (saschagrunert) [reply supported_test.go:10]: Yep, added a whole file and the test case.

File: seccomp_linux.go (hunk 1 above)
Comment (saschagrunert) [reply seccomp_linux.go:182]: My main intention was to use the method in a different project without relying on the seccomp build tag. Therefore I'd have a need for this public method :) WDYT?

File: supported.go (hunk 2 above)
Comment (saschagrunert) [reply supported.go:36]: Removed the change as discussion in runc: https://github.com/opencontainers/runc/pull/2555#pullrequestreview-468870158

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**